File: app/src/main/java/com/remotearthsolutions/expensetracker/adapters/AccountListAdapter.java
package com.remotearthsolutions.expensetracker.adapters;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;
import com.remotearthsolutions.expensetracker.R;
import com.remotearthsolutions.expensetracker.adapters.viewholder.AccountViewHolder;
import com.remotearthsolutions.expensetracker.databaseutils.models.dtos.AccountIncome;

import java.util.List;

public class AccountListAdapter extends RecyclerView.Adapter<AccountViewHolder> {

    private List<AccountIncome> accountlist;
    private AccountListAdapter.OnItemClickListener listener;

    public AccountListAdapter(List<AccountIncome> accountlist) {
        this.accountlist = accountlist;
    }

    @Override
    public AccountViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_account, parent, false);
        return new AccountViewHolder(v, listener);
    }

    @Override
    public void onBindViewHolder(AccountViewHolder holder, int position) {
        AccountIncome account = accountlist.get(position);
        holder.bind(account);
    }

    @Override
    public int getItemCount() {

        if (accountlist == null) return 0;
        return accountlist.size();
    }

    public void setOnItemClickListener(AccountListAdapter.OnItemClickListener listener) {
        this.listener = listener;
    }

    public interface OnItemClickListener {
        void onItemClick(AccountIncome account);
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/adapters/AccountListAdapter.kt
package com.remotearthsolutions.expensetracker.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.remotearthsolutions.expensetracker.R
import com.remotearthsolutions.expensetracker.adapters.viewholder.AccountViewHolder
import com.remotearthsolutions.expensetracker.databaseutils.models.AccountModel

class AccountListAdapter(private val accountList: List<AccountModel>?) :
    RecyclerView.Adapter<AccountViewHolder>() {
    private var listener: OnItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AccountViewHolder {
        val v =
            LayoutInflater.from(parent.context).inflate(R.layout.custom_account, parent, false)
        return AccountViewHolder(v, listener!!)
    }

    override fun onBindViewHolder(holder: AccountViewHolder, position: Int) {
        val account = accountList!![position]
        holder.bind(account)
    }

    override fun getItemCount(): Int {
        return accountList?.size ?: 0
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        this.listener = listener
    }

    interface OnItemClickListener {
        fun onItemClick(account: AccountModel)
    }

}